File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] result = new int[2];
        int l=0, r=numbers.length-1;
        while (l<r) {
            int left = numbers[l];
            int right = numbers[r];
            if (left + right == target) {
                result[0] = l+1;
                result[1] = r+1;
                break;
            } else if (left + right < target) {
                l++;
            } else {
                r--;
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        counts = [0] * 26
        left = 0
        max_length = 0
        max_frequency = 0
        for right, ch in enumerate(s):
            counts[ord(ch)-ord('A')] += 1
            max_frequency = max(max_frequency, counts[ord(ch)-ord('A')])
            while (right - left + 1 - max_frequency) > k:
                counts[ord(s[left])-ord('A')] -= 1
                max_frequency = max(max_frequency, counts[ord(s[left])-ord('A')])
                left += 1
            max_length = max(max_length, right-left+1)
        return max_length